inst/resources/lua/uri-to-fn.lua

```diff
@@ -0,0 +1,35 @@
+local options = {} -- where we store pandoc Meta
+
+local function getMeta(meta)
+  options = meta
+  return nil -- Do not modify Meta
+end
+
+local function uriToFootnote(link)
+  local target = link.target
+  local content = link.content
+  local contentString = pandoc.utils.stringify(content)
+  local footnoteContent
+  local attr = pandoc.Attr("", {}, {})
+
+  if not options["links-to-footnotes"] then return nil end
+  if target == contentString then return nil end
+  if target == "mailto:" .. contentString then return nil end
+
+  if string.find(target, "://") then
+    footnoteContent = {pandoc.Para({pandoc.Link(target, target, "", attr)})}
```

Comment:
For such links, we need to add the class `uri` to prevent duplicate URLs from appearing after the links:
https://github.com/rstudio/pagedown/blob/a9e6f1e2f265e47c8e506c9ed6bcc03d5fda8627/inst/resources/css/default.css#L111-L112

![image](https://user-images.githubusercontent.com/163582/47406521-5c284800-d71c-11e8-98c6-505d48b948fd.png)

Reply:
Good catch! I was first surprised because I've checked this point. But I forgot to test on different versions of Pandoc (I developed and tested with Pandoc < 2.3.1).  
The point is that in Pandoc 2.3.1, the `uri` class is added in the markdown reader and no longer in the HTML writer, see https://pandoc.org/releases.html#pandoc-2.3.1-28-september-2018.  
I'll correct the filter... and update my Pandoc version ;).